sample_size/scripts/input_utils.py

```diff
@@ -80,9 +78,43 @@ def get_metric_parameters(parameter_definitions: Dict[str, str]) -> Dict[str, fl
     return parameters
 
 
-def get_metric_metadata() -> Tuple[str, Dict[str, float]]:
-    metric_type = get_metric_type().lower()
-    metric_metadata = get_metric_parameters(METRIC_PARAMETERS[metric_type])
-    metric_metadata["mde"] = get_mde(metric_type)
+def get_variants() -> int:
+    number_of_variants = (
+        input(
+            "Enter the number of cohorts for this test or Press Enter to use default variant = 2 if you have only 1 "
+            "control and 1 treatment. \n"
+            "definition: Control + number of treatments: "
+        )
+        .strip()
+        .lower()
+    )
+    if number_of_variants.isdigit():
+        if int(number_of_variants) < 2:
+            raise ValueError("Error: An experiment must contain at least 2 variants.")
+        return int(number_of_variants)
+    elif number_of_variants == "":
+        print("Using default variants(2)...")
+        return 2
+    else:
+        raise ValueError("Error: Please enter a positive integer for the number of variants.")
+
+
+def register_another_metric() -> bool:
+    register = input("Are you going to register another metric? (y/n)").strip().lower()
+    if register == "y":
+        return True
+    elif register in ["n", ""]:
+        return False
+    else:
+        raise ValueError("Error: Please enter 'y' or 'n'.")
+
+
+def get_metrics() -> List[Dict[str, Any]]:
+    metrics = []
+    while register_another_metric():
```

Comment:
> # TODO: add the case of registering the first metric
This can be handled by emulating a do while loop

Reply:
@xyzou685 are you planning to fix this by another PR? I would suggest either making change in this PR or just create another one only for that change. Instead of putting it with the `merge multiple-testing-calculator to main` PR you will make at the end otherwise it will be hard to know the exact new change that is not approved yet.

Reply:
Creating a standalone PR for this TODO makes sense to me. The priority of editing the scripts is low in terms of unblocking the rest of the calculator portal tickets.

Reply:
it is probably better to release together? i don't think that TODO is hard to block the other work for so long though. just thinking we can then make the release once instead of making twice just due to that small TODO

Reply:
Do you mean fixing it in this PR? I don't see a straightforward way to fix it since we cannot check whether the metric being registered is the first metric and customize questions based on that. If we simply want it to work properly can I just revert my previous change on this for now https://github.com/godaddy/sample-size/pull/30#discussion_r882216839, as it actually works, though not in the most efficient way...

Reply:
Why is your original version not efficient? i don't see super big difference though

oh i just mean, either fix it in this PR, or create a new individual PR. Not with the PR which is to merge all your change with `main` branch so we know what is updated for this TODO. My opinion is a bit different with dannie, i guess she suggests you merging all the change and do the release without finishing this TODO to unblock other works. but I think this TODO is not really hard, we can just do it and do one release to support multiple metrics. instead of releasing a version with a broken (incomplete) script.

Reply:
or you can do something similar to what Dileep has suggested:
e.g.
```python
def get_metrics() -> List[Dict[str, Any]]:
    metrics = []
    def get_metric():
        metric_type = get_metric_type()
        metric_metadata = get_metric_parameters(METRIC_PARAMETERS[metric_type])
        metric_metadata["mde"] = get_mde(metric_type)
        metrics.append({"metric_type": metric_type, "metric_metadata": metric_metadata})

    get_metric()
    while register_another_metric():
        get_metric()

    return metrics
```